lib/matplotlib/offsetbox.py

```diff
@@ -40,6 +42,25 @@
 DEBUG = False
 
 
+def _compat_get_offset(meth):
+    """
+    Decorator for the get_offset method of OffsetBox and subclasses, that
+    allows supporting both the new signature (self, bbox, renderer) and the old
+    signature (self, width, height, xdescent, ydescent, renderer).
+    """
+    sigs = [lambda self, width, height, xdescent, ydescent, renderer: locals(),
+            lambda self, bbox, renderer: locals()]
+
+    @functools.wraps(meth)
+    def get_offset(self, *args, **kwargs):
+        params = _api.select_matching_signature(sigs, self, *args, **kwargs)
```

Comment:
Should probably warn if the old signature is used, `select_matching_signature` does not do so itself.

Reply:
I intentionally chose not to deprecate anything yet as I think we can live at least with one release supporting both apis to make sure the new one makes sense?